Assignment-11-September-4/vite-project/src/store/loginThroughCodeSlice.tsx

```diff
@@ -0,0 +1,70 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import emailjs from "@emailjs/browser";
+import axios from "axios";
+import { BASE_URL } from "../Utilities/baseURL";
+import { AxiosResponse } from "axios";
+
+const checkUserExists = async (email: string): Promise<any> => {
+  try {
+    const response: AxiosResponse = await axios.get(
+      `${BASE_URL}/users?email=${email}`
+    );
+    return response.data[0] || null;
+  } catch (error) {
+    throw new Error("Failed to check user existence");
+  }
+};
+
+export const sendVerificationEmail = createAsyncThunk(
+  "auth/sendVerificationEmail",
+  async (email: string, { rejectWithValue }) => {
+    try {
+      const user = await checkUserExists(email);
+
+      if (!user) {
+        return rejectWithValue("User does not exist");
+      }
+
+      const emailParams = {
+        to_name: email,
+        from_name: "SurveyCopsTeam",
+        message: `Your user details: ${JSON.stringify(user)}`,
+      };
+
+      await emailjs.send(
+        "service_hy6uao9",
+        "template_310losc",
+        emailParams,
+        "IZwPmNZdJoi2j0ttx"
```

Comment:
is this some kind of personalize token? if yes then we shouldn't push it in to the code . it always come from .env